components/brave_new_tab_ui/reducers/new_tab_reducer.ts

```diff
@@ -179,6 +179,17 @@ export const newTabReducer: Reducer<NewTab.State | undefined> = (state: NewTab.S
       }
       break
 
+    case types.UPDATE_CLOCK_WIDGET: {
+      const { showClockWidget, clockFormat } = payload
+      preferencesAPI.saveShowClock(showClockWidget)
+      preferencesAPI.saveClockFormat(clockFormat)
```

Comment:
Please wrap these non-reducer lines with `performSideEffect(() => { ...[here]... })` (same with the grid sites PR, by the way). This ensures that anything that may be synchronous between JS and Chromium is not delaying the reducer from completing.

Reply:
done!